### PR TITLE
(SERVER-2801) Update Bouncy Castle to 1.66

### PR DESCRIPTION
**Pull request by Magisus**

### CODE REVIEW

File: project.clj [hunk 1]
```diff
@@ -136,7 +136,7 @@
                          [org.bouncycastle/bcpkix-fips "1.0.3"]
                          [org.bouncycastle/bc-fips "1.0.2"]
                          [org.bouncycastle/bctls-fips "1.0.10"]
-                         [org.bouncycastle/bcpkix-jdk15on "1.60"]]
+                         [org.bouncycastle/bcpkix-jdk15on "1.65"]]
```
Comment (mwaggett) [inline project.clj:139]: I think latest is 1.66 now

Comment (Magisus) [reply project.clj:139]: Oh. Hm. Might need to re-test then.

Comment (Magisus) [reply project.clj:139]: `ssl-utils` was green, doing server now.

**Comment by mwaggett** [comment]
From the email Justin got back from bouncycastle:

> Note there have been updates to the other FIPS APIs as well (these are also mentioned in the release notes). If you're doing a round of updates at the moment it might be worth hanging back for a week as we'll be publishing the other APIs soon.

I haven't seen any updates yet (this was sent 9 days ago), not sure if we want to wait for these before releasing? Mostly I worry that we'll forget to check back, but maybe that is a separate concern.

File: project.clj (hunk 1 above)
Comment (Magisus) [reply project.clj]: Server is also green.

**Comment by Magisus** [comment]
I think I'd rather get this and integrating, so we can catch any other bugs that might be hiding. I'll file a separate ticket about the FIPS libs.

**Comment by Magisus** [comment]
This was tested in PDB and RBAC, successfully.

**Pull request merged**